go/teams/chain.go

```diff
@@ -1594,9 +1628,43 @@ func (t *teamSigchainPlayer) addInnerLink(
 			return res, fmt.Errorf("unsupported link type: %s", payload.Body.Type)
 		}
 	}
-	if isHighLink {
-		res.newState.inner.LastHighLinkID = link.LinkID().Export()
-		res.newState.inner.LastHighSeqno = link.Seqno()
+
+	hPrevInfo := payload.HPrevInfo
+	reverifyNeeded := res.newState.ReverifyNeededForHighSkips()
+	highSkipVerifyDesired := !isInflate && !precheck
+	// If presented with an hPrevInfo, and we are unable to verify it,
+	// but would like to, throw a GreenLinkError to force a repoll.
+	if hPrevInfo != nil && highSkipVerifyDesired && reverifyNeeded {
+		fmt.Printf("THROWING GREEN LINK ERROR! %v\n", payload.Seqno)
+		return res, GreenLinkError{msg: "Did not have information to compute expected team high skip; reverifying."}
+	}
+	// We only run addInnerLink out of order when inflating, and so we skip
+	// computing high skips during it.
+	// updates high prev data, so we skip it. If hPrevInfo is presented and
```

Comment:
this comment could be clearer if it didn't look like it was mid-way through a copy-paste job